### PR TITLE
Refactor Languages and Reference

### PR DESCRIPTION
**Pull request by plaindocs**

### CODE REVIEW

File: user/languages/ruby.md [hunk 1]
```diff
@@ -7,104 +7,62 @@ layout: en
...
-## Supported Ruby Versions and RVM
-
-The Ruby VM provides recent versions of:
-
-- Ruby: 2.2.0, 2.1.x, 2.0.0, 1.9.3, 1.9.2 and 1.8.7
-- JRuby: 1.7.x (1.8 and 1.9 mode)
-- Ruby Enterprise Edition: 1.8.7 2012.02
-
-Pre-compiled versions are downloaded on demand from our [Ruby version cache](http://rubies.travis-ci.org).
+> We're moving all version numbers and other distro specific information over to
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: I think I'd make this more generic, like: 

> Check language versions pre-installed and other build-environment specific information in our reference pages at:

Thoughts?

File: user/languages/ruby.md [hunk 2]
```diff
@@ -221,36 +152,24 @@ bundler_args: --without production
...
 
-You can specify a custom Gemfile name:
+Specify a custom Gemfile name:
 
 ```yaml
 gemfile: gemfiles/Gemfile.ci
 ```
 
-Unless specified, the worker will look for a file named "Gemfile" in the root of
+Unless specified, the build will look for a file named `Gemfile` in the root of
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: yay to common terms!

File: user/languages/ruby.md [hunk 3]
```diff
@@ -221,36 +152,24 @@ bundler_args: --without production
...
-You can also set [extra arguments](http://bundler.io/v1.3/man/bundle-install.1.html)
-extra arguments to be passed to `bundle install`:
+You can pass [extra arguments](http://bundler.io/v1.3/man/bundle-install.1.html)
+ to `bundle install`:
 
 ```yaml
 bundler_args: --binstubs
 ```
 
-You can also define a script to be run before 'bundle install':
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: I think maybe this was useful?

File: user/languages/ruby.md [hunk 4]
```diff
@@ -259,19 +178,10 @@ HAML, Sinatra, Ruby on Rails,etc.
 To test against multiple versions of dependencies:
 
 1. Create a directory in your project's repository root where you will keep
-   gemfiles (./gemfiles is a commonly used name)
-2. Add one or more gemfiles to it
-3. Instruct Travis CI to use those gemfiles using the *gemfile* option in your
-   .travis.yml
-
-For example, amqp gem is [tested against EventMachine 0.12.x and 1.0
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: While this is old-ish, seems the example to specify different Gemfiles, like gemfiles/eventmachine-pre linking to ruby-amqp, might still be relevant and helpful?

File: user/languages/ruby.md [hunk 5]
```diff
@@ -310,44 +220,28 @@ engines, hosted service providers and so on.
...
-- Oracle JDK 8
-- OpenJDK 6
-
-To do so, use the `jdk` key in your `.travis.yml`, for example:
+Test projects against multiple JDKs, by using the `jdk` key in your
+`.travis.yml`:
 
 ```yaml
 jdk:
   - oraclejdk7
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: Given that we're removing the section about all 4, I'd add to this list `- oraclejdk8` .  What do you think?

File: user/reference/osx.md [hunk 6]
```diff
@@ -1,7 +1,8 @@
 ---
 title: The OS X Build Environment
 layout: en
-
+redirect_from:
+  - /user/osx-ci-environment/
```
Comment (MariadeAnton) [inline user/reference/osx.md]: noice! 🎉

File: user/languages/ruby.md (hunk 3 above)
Comment (plaindocs) [inline user/languages/ruby.md]: Kinda, but it is't a Ruby thing. It is a normal build config thing.

File: user/languages/ruby.md (hunk 1 above)
Comment (plaindocs) [inline user/languages/ruby.md]: How about: 

> Language versions and other build-environment specific
> information are in our reference pages:
>  * [Precise](/user/reference/precise/)
>  * [Trusty](/user/reference/trusty/)
>  * [OS X](/user/reference/osx/)

The rest of this guide covers configuring Ruby projects on Travis CI. If you're
new to Travis CI please read our [Getting Started](/user/getting-started/) and
[build configuration](/user/customizing-the-build/) guides first.

File: user/languages/ruby.md (hunk 4 above)
Comment (plaindocs) [inline user/languages/ruby.md]: Yeah, I left the others in. Should be enough I think?

**Comment by plaindocs** [comment]
Also, I have some link fixups to do.

**Comment by plaindocs** [comment]
Well *THAT* turned into a bit of a rabbit hole...

**Comment by MariadeAnton** [comment]
This is looking great, @plaindocs - thanks a bunch for all the fixes and links edits! 💯 

Would it make sense to squash this commits into just one or two of them?

Also, would `sw-refactor-trusty` be the "Trusty Default release" we should point all future updates towards? Thank you!

File: user/languages/ruby.md [hunk 7]
```diff
@@ -19,6 +19,27 @@ The rest of this guide covers configuring Ruby projects on Travis CI. If you're
...
 
+<div markdown="block" class="ataglance">
+
+|                   | Default                                   |
+|:------------------|:------------------------------------------|
+| Typical `install` | `bundle install --jobs=3 --retry=3`       |
+| Typical `script`  | `rake`                                    |
+| Matrix keys       | `env`, `rvm`, `gemfile`, `jdk`            |
+| Support           | [Travis CI](mailto:support@travis-ci.com) |
+
```
Comment (MariadeAnton) [inline user/languages/ruby.md]: 💃

File: user/reference/precise.md [hunk 8]
```diff
@@ -106,8 +107,8 @@ images.
 
 For other images, see the list below:
```
Comment (MariadeAnton) [inline user/reference/precise.md]: images vs build environments?

File: user/reference/precise.md [hunk 9]
```diff
@@ -419,17 +420,23 @@ Python 2.4 and Jython *are not supported* and there are no plans to support them
 
 On all versions except pypy and pypy3 have `numpy` as well.
 
-## Ruby (aka common) VM images
+## Ruby images
 
-### Ruby versions/implementations
+The Ruby images contain recent versions of:
```
Comment (MariadeAnton) [inline user/reference/precise.md]: This refers to Precise, right?

File: user/reference/precise.md (hunk 8 above)
Comment (plaindocs) [inline user/reference/precise.md]: YES! There is a *lot* of fix up todo there though.

**Comment by plaindocs** [comment]
Thanks <3 for the good words!

* [ ] , I'll pull the current table out of Precise, figure out exactly how we're gonna update with regards to `amethyst` etc
* [ ] fixup for images / environments

**Pull request merged**